### ISSUE TITLE
Unknown layer type: DepthwiseConvolution

### ISSUE THREAD

**Issue by FSetragno** [imistyrain/MobileFaceNet#13]
While trying to train the network on a custom dataset, I get the following error:

F0731 09:17:29.721236  8859 layer_factory.hpp:81] Check failed: registry.count(type) == 1 (0 vs. 1) **Unknown layer type: DepthwiseConvolution** (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm, BatchReindex, Bias, Concat, ContrastiveLoss, Convolution, Crop, Data, Deconvolution, Dropout, DummyData, ELU, Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, HDF5Data, HDF5Output, HingeLoss, Im2col, ImageData, InfogainLoss, InnerProduct, Input, LRN, LSTM, LSTMUnit, MVN, MarginInnerProduct, MemoryData, MultinomialLogisticLoss, PReLU, Parameter, Pooling, Power, RNN, ReLU, Reduction, Reshape, SPP, Scale, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, Softmax, SoftmaxWithLoss, Split, TanH, Threshold, Tile, WindowData)

**Comment by imistyrain** [comment]
You can find necessaries files in [here](https://github.com/yonghenglh6/DepthwiseConvolution/tree/master/caffe). Copy them to your caffe dir and build it.
Note: you should recompile caffe to use it.
